[**@adcp/client API Reference v2.0.0**](../README.md)

***

[@adcp/client API Reference](../README.md) / TaskAbortedError

# Class: TaskAbortedError

Defined in: [src/lib/errors/index.ts:58](https://github.com/adcontextprotocol/adcp-client/blob/9ed0be764adbd110916d257101c95a577b3f15c8/src/lib/errors/index.ts#L58)

Error thrown when a task is aborted

## Extends

- [`ADCPError`](ADCPError.md)

## Constructors

### Constructor

> **new TaskAbortedError**(`taskId`, `reason?`): `TaskAbortedError`

Defined in: [src/lib/errors/index.ts:61](https://github.com/adcontextprotocol/adcp-client/blob/9ed0be764adbd110916d257101c95a577b3f15c8/src/lib/errors/index.ts#L61)

#### Parameters

##### taskId

`string`

##### reason?

`string`

#### Returns

`TaskAbortedError`

#### Overrides

[`ADCPError`](ADCPError.md).[`constructor`](ADCPError.md#constructor)

## Properties

### details?

> `optional` **details**: `any`

Defined in: [src/lib/errors/index.ts:9](https://github.com/adcontextprotocol/adcp-client/blob/9ed0be764adbd110916d257101c95a577b3f15c8/src/lib/errors/index.ts#L9)

#### Inherited from

[`ADCPError`](ADCPError.md).[`details`](ADCPError.md#details)

***

### code

> `readonly` **code**: `"TASK_ABORTED"` = `'TASK_ABORTED'`

Defined in: [src/lib/errors/index.ts:59](https://github.com/adcontextprotocol/adcp-client/blob/9ed0be764adbd110916d257101c95a577b3f15c8/src/lib/errors/index.ts#L59)

#### Overrides

[`ADCPError`](ADCPError.md).[`code`](ADCPError.md#code)

***

### taskId

> `readonly` **taskId**: `string`

Defined in: [src/lib/errors/index.ts:62](https://github.com/adcontextprotocol/adcp-client/blob/9ed0be764adbd110916d257101c95a577b3f15c8/src/lib/errors/index.ts#L62)

***

### reason?

> `readonly` `optional` **reason**: `string`

Defined in: [src/lib/errors/index.ts:63](https://github.com/adcontextprotocol/adcp-client/blob/9ed0be764adbd110916d257101c95a577b3f15c8/src/lib/errors/index.ts#L63)

***

### stackTraceLimit

> `static` **stackTraceLimit**: `number`

Defined in: node\_modules/@types/node/globals.d.ts:68

The `Error.stackTraceLimit` property specifies the number of stack frames
collected by a stack trace (whether generated by `new Error().stack` or
`Error.captureStackTrace(obj)`).

The default value is `10` but may be set to any valid JavaScript number. Changes
will affect any stack trace captured _after_ the value has been changed.

If set to a non-number value, or set to a negative number, stack traces will
not capture any frames.

#### Inherited from

[`ADCPError`](ADCPError.md).[`stackTraceLimit`](ADCPError.md#stacktracelimit)

***

### cause?

> `optional` **cause**: `unknown`

Defined in: node\_modules/typescript/lib/lib.es2022.error.d.ts:26

#### Inherited from

[`ADCPError`](ADCPError.md).[`cause`](ADCPError.md#cause)

***

### name

> **name**: `string`

Defined in: node\_modules/typescript/lib/lib.es5.d.ts:1076

#### Inherited from

[`ADCPError`](ADCPError.md).[`name`](ADCPError.md#name)

***

### message

> **message**: `string`

Defined in: node\_modules/typescript/lib/lib.es5.d.ts:1077

#### Inherited from

[`ADCPError`](ADCPError.md).[`message`](ADCPError.md#message)

***

### stack?

> `optional` **stack**: `string`

Defined in: node\_modules/typescript/lib/lib.es5.d.ts:1078

#### Inherited from

[`ADCPError`](ADCPError.md).[`stack`](ADCPError.md#stack)

## Methods

### captureStackTrace()

> `static` **captureStackTrace**(`targetObject`, `constructorOpt?`): `void`

Defined in: node\_modules/@types/node/globals.d.ts:52

Creates a `.stack` property on `targetObject`, which when accessed returns
a string representing the location in the code at which
`Error.captureStackTrace()` was called.

```js
const myObject = {};
Error.captureStackTrace(myObject);
myObject.stack;  // Similar to `new Error().stack`
```

The first line of the trace will be prefixed with
`${myObject.name}: ${myObject.message}`.

The optional `constructorOpt` argument accepts a function. If given, all frames
above `constructorOpt`, including `constructorOpt`, will be omitted from the
generated stack trace.

The `constructorOpt` argument is useful for hiding implementation
details of error generation from the user. For instance:

```js
function a() {
  b();
}

function b() {
  c();
}

function c() {
  // Create an error without stack trace to avoid calculating the stack trace twice.
  const { stackTraceLimit } = Error;
  Error.stackTraceLimit = 0;
  const error = new Error();
  Error.stackTraceLimit = stackTraceLimit;

  // Capture the stack trace above function b
  Error.captureStackTrace(error, b); // Neither function c, nor b is included in the stack trace
  throw error;
}

a();
```

#### Parameters

##### targetObject

`object`

##### constructorOpt?

`Function`

#### Returns

`void`

#### Inherited from

[`ADCPError`](ADCPError.md).[`captureStackTrace`](ADCPError.md#capturestacktrace)

***

### prepareStackTrace()

> `static` **prepareStackTrace**(`err`, `stackTraces`): `any`

Defined in: node\_modules/@types/node/globals.d.ts:56

#### Parameters

##### err

`Error`

##### stackTraces

`CallSite`[]

#### Returns

`any`

#### See

https://v8.dev/docs/stack-trace-api#customizing-stack-traces

#### Inherited from

[`ADCPError`](ADCPError.md).[`prepareStackTrace`](ADCPError.md#preparestacktrace)
